### PR TITLE
bluetooth: fast_pair: Add TX power calibration documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces documentation related to TX power calibration.

Jira: NCSDK-16584

### CODE REVIEW

**Review by ahasztag — APPROVED** [review]

**Review by jciupis — APPROVED** [review]

File: doc/nrf/ug_bt_fast_pair.rst [hunk 1]
```diff
@@ -93,6 +98,31 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
+* provided during model registration,
+* included in the advertising payload.
+
+Advertising data provider
+-------------------------
+
+If your application uses :ref:`bt_le_adv_prov_readme`, you can use the TX power advertising data provider (:kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER`) to read the advertising TX power from Bluetooth controller and add it to the generated advertising data.
+The :kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER_CORRECTION_VAL` option can be used to define a TX power correction value that is added to the TX power readout included in the advertising data.
+The option can be used to take into account hardware configuration, for example used antenna and device casing.
+See :ref:`peripheral_fast_pair` sample for an example of the TX power advertising provider usage.
```
Comment (divipillai) [inline doc/nrf/ug_bt_fast_pair.rst:116]:
```suggestion
The option can be used to take into account hardware configuration, for example, used antenna and device casing.
See :ref:`peripheral_fast_pair` sample for an example of how to use the TX power advertising provider.
```

File: doc/nrf/ug_bt_fast_pair.rst [hunk 2]
```diff
@@ -93,6 +98,31 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
+
+Multiprotocol Service Layer Front-End Module (MPSL FEM)
+-------------------------------------------------------
+
+If your application uses MPSL :ref:`nrfxlib:mpsl_fem`, you can use a Frond-End Module power model.
+The power model allows you to control the TX power more accurately and compensate, for example, for external conditions.
+See the ``TX power split using models`` section of :ref:`mpsl_fem` documentation for details.
```
Comment (divipillai) [inline doc/nrf/ug_bt_fast_pair.rst:123]:
```suggestion
See the TX power split using models section of the :ref:`mpsl_fem` documentation for more details.
```

File: doc/nrf/ug_bt_fast_pair.rst [hunk 3]
```diff
@@ -93,6 +98,31 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
 See :file:`samples/bluetooth/peripheral_fast_pair/src/bt_adv_helper.c` for an example of the implementation.
 
+Advertising TX power
+====================
+
+According to the `Fast Pair TX power`_ documentation, the Fast Pair Seeker must know the Provider's TX power to determine proximity.
+The TX power can be either:
+
+* provided during model registration,
+* included in the advertising payload.
```
Comment (divipillai) [inline doc/nrf/ug_bt_fast_pair.rst:108]:
```suggestion
The Fast Pair Seeker must know the TX power of the provider to determine proximity. 
The TX power can be included in one of the following ways:

* During model registration
* Advertising payload

See the `Fast Pair TX power`_ documentation for more information.
```

File: doc/nrf/ug_bt_fast_pair.rst [hunk 4]
```diff
@@ -93,6 +98,31 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
+
+Advertising data provider
+-------------------------
+
+If your application uses :ref:`bt_le_adv_prov_readme`, you can use the TX power advertising data provider (:kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER`) to read the advertising TX power from Bluetooth controller and add it to the generated advertising data.
+The :kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER_CORRECTION_VAL` option can be used to define a TX power correction value that is added to the TX power readout included in the advertising data.
+The option can be used to take into account hardware configuration, for example used antenna and device casing.
+See :ref:`peripheral_fast_pair` sample for an example of the TX power advertising provider usage.
+
+Multiprotocol Service Layer Front-End Module (MPSL FEM)
```
Comment (divipillai) [inline doc/nrf/ug_bt_fast_pair.rst:118]:
```suggestion
Multiprotocol Service Layer front-end module (MPSL FEM)
```

File: doc/nrf/ug_bt_fast_pair.rst [hunk 5]
```diff
@@ -93,6 +98,31 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
+-------------------------------------------------------
+
+If your application uses MPSL :ref:`nrfxlib:mpsl_fem`, you can use a Frond-End Module power model.
```
Comment (divipillai) [inline doc/nrf/ug_bt_fast_pair.rst:121]:
```suggestion
If your application uses MPSL :ref:`nrfxlib:mpsl_fem`, you can use a front-end module power model.
```

File: doc/nrf/ug_bt_fast_pair.rst (hunk 3 above)
Comment (MarekPieta) [reply doc/nrf/ug_bt_fast_pair.rst]: Rephrased this one a bit

File: doc/nrf/ug_bt_fast_pair.rst [hunk 6]
```diff
@@ -93,6 +98,33 @@ The Bluetooth privacy is selected by the Fast Pair service, but you must make su
...
 See :file:`samples/bluetooth/peripheral_fast_pair/src/bt_adv_helper.c` for an example of the implementation.
 
+Advertising TX power
+====================
+
+The Fast Pair Seeker must know the TX power of the Provider to determine proximity.
+The TX power can be provided in one of the following ways:
+
+* defined during model registration
+* included in the advertising payload
```
Comment (annwoj) [inline doc/nrf/ug_bt_fast_pair.rst:108]:
```suggestion
* Defined during model registration
* Included in the advertising payload
```

**Review by alstrzebonski — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Comment by MarekPieta** [comment]
Rebased onto main

**Comment by MarekPieta** [comment]
Updated `west.yml` after changes were merged to `sdk-nrfxlib`

**Pull request merged**